### PR TITLE
pmemcheck: remove/reserve unused macros

### PR DESCRIPTION
**Pull request by wlemkows**
This is part of changes related to new VALGRIND_EMIT_LOG macro.
This macro allows emit any text value to output store log file.

To fully integrate all changes into Valgrind and pmdk
the following steps in specific order are required:

1. pmemcheck: add VALGRING_EMIT_LOG support
https://github.com/pmem/valgrind/pull/58 **(DONE)**

2. pmdk: trigger docker update (upgrade valgrind)
https://github.com/pmem/pmdk/pull/3096  **(DONE)**

___________________________________________________
time to upgrade valgrind on machines **(DONE)**
___________________________________________________

3. pmdk: add VALGRING_EMIT_LOG support
https://github.com/pmem/pmdk/pull/3094 **(DONE)**

4. pmdk: remove old macros replaced by VALGRIND_EMIT_LOG
https://github.com/pmem/pmdk/pull/3098  **(DONE)**

5. pmemcheck: remove old macros replaced by VALGRIND_EMIT_LOG
https://github.com/pmem/valgrind/pull/59

6. pmdk: trigger docker update (upgrade valgrind)
TODO

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/valgrind/59)

### CODE REVIEW

File: pmemcheck/pmc_main.c [hunk 1]
```diff
@@ -1753,41 +1742,6 @@ pmc_handle_client_request(ThreadId tid, UWord *arg, UWord *ret )
             break;
         }
 
-        case VG_USERREQ__PMC_FULL_REORDED: {
```
Comment (marcinslusarz) [inline pmemcheck/pmc_main.c:1756]: Please add switch for reserved values with an error message, just in case of someone messing up theirs environment.

**Comment by marcinslusarz** [comment]
Does it really depend on pmem/pmdk#3098?

**Comment by wlemkows** [comment]
Hmm, my mistake, it does not depend on pmem/pmdk#3098.

File: pmemcheck/pmc_main.c (hunk 1 above)
Comment (wlemkows) [reply pmemcheck/pmc_main.c:1756]: Done.

**Review by marcinslusarz — APPROVED** [review]

**Pull request merged**